test/unit/argument_parser/argument_parser_design_error_test.cpp

```diff
@@ -44,34 +44,41 @@ TEST(parse_test, design_error)
     EXPECT_THROW(parser3.add_option(option_value, '\0', "", "oh oh all is empty."),
                  seqan3::design_error);
 
-    bool flag_value;
+    bool true_value{true};
 
-    // short flag
+    // default true
     seqan3::argument_parser parser4{"test_parser", 1, argv};
+    EXPECT_THROW(parser4.add_flag(true_value, 'i', "int", "oh oh default is true."),
+                 seqan3::design_error);
+
+    bool flag_value{false};
+
+    // short flag
+    seqan3::argument_parser parser5{"test_parser", 1, argv};
     parser4.add_flag(flag_value, 'i', "int1", "this is an int option.");
```

Comment:
I think this line throws.

Change in line 27:

```cpp
int option_value;
```

to

```cpp
int option_value{};
```

If it's not initialised, its value is basically whatever was in that memory location before (it's almost certain not `0`). It then gets converted to `bool`. If `option_value` is anything but `0`, it will be converted to `true`.

Reply:
But 
```c++
int option_value;  
```
is only used for the options and not flags. I don't think it would be converted to `bool`